zjit/src/cruby_methods.rs

```diff
@@ -59,10 +66,66 @@ fn annotate_c_method(props_map: &mut HashMap<*mut c_void, FnProperties>, class:
     props_map.insert(fn_ptr, props);
 }
 
+/// Look up a method and find its builtin function pointer by parsing its ISEQ
+fn annotate_builtin_method(props_map: &mut HashMap<*mut c_void, FnProperties>, class: VALUE, method_name: &'static str, props: FnProperties) {
+    unsafe {
```

Comment:
@k0kubun I used `rb_yjit_constant_ic_update` as a reference for this and I see it has `with_vm_lock` before going through the iseq. Is it safe to not have it here given that it's executed at compile time?

Reply:
> cruby_methods::init() is called in ZJITState::init() during boot.

🤦‍♂️ I confused myself about the macro/function's execution time. 

> Even when we do, I don't think it's going to be this function's responsibility to take the VM lock; we will need it for the whole rb_zjit_init().

Thanks for clarifying it